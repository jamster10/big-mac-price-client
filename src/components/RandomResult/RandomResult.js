import React from 'react'
import './RandomResult.css';


export default function RandomResult({currentCountry, randomCountry, amount}) {
  const bigMacCount = () => {
    const count = (amount / randomCountry.local_price) * (currentCountry.local_price / randomCountry.dollar_price)
    return `${count.toFixed(3)} Big Mac${count > 1 ? 's' : ''}`
  }
  
  const currencyValue = () => {
    const value = amount * (currentCountry.local_price / randomCountry.dollar_price)
    return `${value.toFixed(3)} dollar${value > 1 ? 's' : ''}`
  }

  return (
    <div>
      <h2>Random Country: {randomCountry.country}</h2>
      <p>You could buy {bigMacCount()} in {randomCountry.country} with {amount}!</p>
      <p>Your {amount} is worth about {currencyValue()} in {randomCountry.country}.</p>
    </div>
  )
}

RandomResult.defaultProps = {
  currentCountry: {
    dollar_price: null,
    local_price: null,
  },
  randomCountry: {
    dollar_price: null,
    local_price: null,
  },
  amount: null
}