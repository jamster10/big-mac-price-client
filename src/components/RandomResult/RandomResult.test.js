import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import RandomResult from './RandomResult';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<RandomResult />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
  const tree = renderer
    .create(<RandomResult/>)
    .toJSON();
  expect(tree).toMatchSnapshot();  
});
