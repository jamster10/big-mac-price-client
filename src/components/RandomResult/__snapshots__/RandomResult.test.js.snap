// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div>
  <h2>
    Random Country: 
  </h2>
  <p>
    You could buy 
    NaN Big Mac
     in 
     with 
    !
  </p>
  <p>
    Your 
     is worth about 
    NaN dollar
     in 
    .
  </p>
</div>
`;
